public/gameInfo.php

```diff
@@ -300,32 +300,133 @@ function drawCharts() {
+
+                    // We need to also keep tracked of `largestWonByCount`, which is later used for chart
+                    // configuration, such as determining the number of gridlines to show.
+                    $currentTotal = $buckets[$bucketIndex]['hardcore'] + $buckets[$bucketIndex]['softcore'];
+                    $largestWonByCount = max($currentTotal, $largestWonByCount);
                 }
-                $wonByUserCount = $achDist[$i];
 
-                if ($wonByUserCount > $largestWonByCount) {
-                    $largestWonByCount = $wonByUserCount;
+                return [$isBucketingEnabled, $buckets, $bucketSize, $largestWonByCount];
+            }
+
+            function handleAllAchievementsCase(int $numAchievements, array $achDist, array $achDistHardcore, array &$buckets): int {
+                if ($numAchievements <= 0) {
+                    return 0;
+                }
+
+                // Add a bucket for the users who have earned all achievements.
+                $buckets[] = [
+                    'hardcore' => $achDistHardcore[$numAchievements],
+                    'softcore' => $achDist[$numAchievements] - $achDistHardcore[$numAchievements],
+                ];
+
+                // Calculate the total count of users who have earned all achievements.
+                // This will later be used for chart configuration in determining the
+                // number of gridlines to show on one of the axes.
+                $allAchievementsCount = (
+                    $achDistHardcore[$numAchievements] + ($achDist[$numAchievements] - $achDistHardcore[$numAchievements])
+                );
+
+                return $allAchievementsCount;
+            }
+
+            function printBucketIteration(int $bucketIteration, int $numAchievements, array $bucket, string $label): void {
+                echo "[ {v:$bucketIteration, f:\"$label\"}, {$bucket['hardcore']}, {$bucket['softcore']} ]";
+            }
+
+            function generateBucketLabelsAndValues(int $numAchievements, array $buckets, int $bucketSize): array {
+                $bucketLabels = [];
+                $hAxisValues = [];
+                $bucketIteration = 0;
+
+                // Loop through each bucket to generate their labels and values.
+                foreach ($buckets as $index => $bucket) {
+                    if ($bucketIteration++ > 0) {
+                        echo ", ";
+                    }
+
+                    // Is this the last bucket? If so, we only want it to include
+                    // players who have earned all the achievements, not a range.
+                    if ($index == count($buckets) - 1) {
+                        $label = "Earned $numAchievements achievements";
+                        printBucketIteration($bucketIteration, $numAchievements, $bucket, $label);
+                        $bucketLabels[] = $label;
+                        $hAxisValues[] = $numAchievements;
```

Comment:
Separating the last item out into its own bucket confused me a bit in the graphs, but I understand the reasoning. From your FFII example:

![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/4906e70b-47a8-458a-b721-afad490f1511)

56-59=4
60-63=3
64-64=1

This is probably related to my other comment about integer-sized buckets.